meerkat/model/tensormodel.py

```diff
@@ -0,0 +1,277 @@
+from __future__ import annotations
+
+import itertools
+from typing import Dict, List
+
+import cytoolz as tz
+import torch
+from tqdm import tqdm
+
+from meerkat import DataPanel
+from meerkat.columns.embedding_column import EmbeddingColumn
+from meerkat.columns.instances_column import InstancesColumn
+from meerkat.columns.prediction_column import ClassificationOutputColumn
+from meerkat.columns.segmentation_column import SegmentationOutputColumn
+from meerkat.columns.tensor_column import TensorColumn
+from meerkat.model.activation import ActivationOp
+from meerkat.model.model import Model
+
+
+class TensorModel(Model):
```

Comment:
I think defining different "kinds" of outputs (classification, time series, etc.) may become a bit unwieldy in this class. It's useful to have those utilities, but in the class definition may not be the right place. Users may have different ways of getting their outputs, so as Sabri mentioned, it may be good to make these methods functions instead.